utilities.c

```diff
@@ -359,3 +365,25 @@ ssize_t __iio_printf iio_snprintf(char *buf, size_t len, const char *fmt, ...)
 
 	return (ssize_t)ret;
 }
+
+uint64_t iio_read_counter_us(void)
+{
+	uint64_t value;
+
+#ifdef _WIN32
+	LARGE_INTEGER freq, cnt;
+
+	QueryPerformanceFrequency(&freq);
+	QueryPerformanceCounter(&cnt);
+
+	value = (1000000 * cnt.QuadPart) / freq.QuadPart;
+#else
+	struct timespec ts;
+
+	clock_gettime(CLOCK_MONOTONIC, &ts);
```

Comment:
It would complicate things quite a bit, I think, if we consider that `iio_read_counter_us` can return an error.

Besides, the Windows implementation is guaranteed not to fail on versions >= XP, and I doubt we want to support anything older than that. And the only way the POSIX implementation can fail, is if monotonic timers are not available, which will never happen on any of the systems that Libiio supports right now.

Reply:
ok, fair enough